pages/user/edit/[user].vue

```diff
@@ -0,0 +1,185 @@
+  const localePath = useLocalePath()
+  const cancel = () => {
+    // if user is a admin
+    if (authStore.hasAccess(['admin'])) {
+      navigateTo(localePath('/admin'))
+    }
+    // else asume normal user
+    navigateTo(localePath('/user'))
+  }
+
+  // submit should update user data
+  const submit = async () => {
+    // if nothing is changed, do nothing
+    if (!hasChanged) {
+      return
+    }
+    // Make a POST request to update the user's data
+    await $fetch('/api/user', {
+      method: 'POST',
+      body: values,
+    })
+    // run cancel function to return to previous page
+    cancel()
+  }
+  // hasChanged is used to check if we have new values
+  const hasChanged = ref(false)
+  watch(values, () => {
+    const raw = toRaw(values)
+    hasChanged.value = JSON.stringify(raw) !== JSON.stringify(initialValues)
+  })
+</script>
+
+<template>
+  <VContainer>
+    <h4 class="font-weight-bold text-center">Edit Profile</h4>
+    <VContainer>
+      <!--Text field to edit name-->
+      <vRow justify="space-around" class="mb-4">
+        <VTextField
+          v-model="values.name"
+          label="Name"
+          type="text"
+          variant="outlined"
+        />
+      </vRow>
+      <!--Text field to edit email-->
+      <vRow justify="space-around" class="mb-4">
+        <VTextField
+          v-model="values.email"
+          label="Email"
+          type="text"
+          variant="outlined"
+        />
+      </vRow>
+
+      <vRow justify="space-around" class="mb-4">
+        <template v-if="authStore.hasAccess(['admin'])">
+          <VMenu transition="slide-y-transition">
+            <!-- activator btn -->
+
+            <template #activator="{ props }">
+              <v-col :cols="6" class="text-center"> User type: </v-col>
+
+              <v-col :cols="6" class="text-center">
+                <VBtn
+                  size="large"
+                  variant="text"
+                  :ripple="false"
+                  :width="'45%'"
+                  v-bind="props"
+                  rounded="lg"
+                  block
+                >
+                  <template #prepend>
+                    {{ values.userType?.toUpperCase() }}
+                  </template>
+                </VBtn>
+              </v-col>
+            </template>
+
+            <VCard
+              class="d-flex flex-column my-2 pa-2"
+              rounded="lg"
+              :elevation="3"
+            >
+              <!-- buttons for usertype -->
+              <VBtn @click="values.userType = 'admin'">Admin</VBtn>
+              <VBtn @click="values.userType = 'company'">Company</VBtn>
+              <VBtn @click="values.userType = undefined">Undefined</VBtn>
+            </VCard>
+          </VMenu>
+        </template>
+      </vRow>
+      <!--Row for buttons-->
+      <vRow justify="space-around" class="mb-4">
+        <!--Submit button-->
+        <v-col :cols="6" class="text-center">
+          <VBtn
```

Comment:
I would switch the order of the buttons to have 'cancel' on the left as in "go back", and 'save' on the right as in "proceed"

Reply:
changed the order, should be fixed